src/metrics/issue-counts.ts

```diff
@@ -0,0 +1,206 @@
+  numOpened: number,
+
+  // How many issues were closed on this day (non-cumulative).
+  numClosed: number,
+};
+
+/**
+ * Issue counts metric result object.
+ */
+export class IssueCountResult {
+  issues: Issue[];
+
+  constructor(issues: Issue[]) {
+    this.issues = issues;
+  }
+
+  /**
+   * Return a time series of issue counts sorted by date, oldest first.
+   */
+  timeSeries(): TimeSeriesPoint[] {
+    const dayPoints = new Map<string, TimeSeriesPoint>();
+
+    function dayPoint(timestamp: string): TimeSeriesPoint {
+      // Issue date from the API is an ISO 8601 timestamp. We just want day
+      // resolution, so take only the YYYY-MM-DD date part.
+      const date = timestamp.split('T')[0];
+      let point = dayPoints.get(date);
+      if (!point) {
+        point = {date, numOpened: 0, numClosed: 0};
+        dayPoints.set(date, point);
+      }
+      return point;
+    }
+
+    for (const issue of this.issues) {
+      dayPoint(issue.openedAt).numOpened++;
+      if (issue.closedAt) {
+        dayPoint(issue.closedAt).numClosed++;
+      }
+    }
+
+    return Array.from(dayPoints.values())
+        .sort((a, b) => a.date.localeCompare(b.date));
+  }
+
+  /**
+   * Return a string summarizing the latest count of open and closed issues.
+   */
+  summary(): string {
+    const total = this.issues.length;
+    let open = 0;
+    let closed = 0;
+    for (const issue of this.issues) {
+      if (issue.closedAt) {
+        closed++;
+      } else {
+        open++;
+      }
+    }
+    return `Found ${total} issues of which ${open} were open and ` +
+        `${closed} were closed.`;
+  }
+}
+
+type Issue = {
+  // YYYY-MM-DDThh:mm:ssZ
+  openedAt: string,
+  closedAt: string,
+};
+
+/**
+ * Fetch all of the issues for a given repo.
+ */
+async function getIssues(
+    github: GitHub, owner: string, name: string): Promise<Issue[]> {
+  let hasNextPage = true;
+  let cursor: string|null = null;
+  const issues: Issue[] = [];
+
+  while (hasNextPage) {
+    const variables: gqlTypes.IssuesQueryVariables = {owner, name, cursor};
+    const result = await github.query<gqlTypes.IssuesQuery>(
+        {query: issuesQuery, variables});
+    const repo = result.data.repository;
+    if (!repo) {
+      break;
+    }
+
+    for (const issue of repo.issues.nodes || []) {
+      if (!issue) {
+        continue;
+      }
+
+      let closedAt = '';
+      if (issue.closed) {
+        for (const ev of issue.timeline.nodes || []) {
+          if (ev && ev.__typename === 'ClosedEvent') {
+            closedAt = ev.createdAt;
+            break;
+          }
+        }
+        if (!closedAt) {
+          // Edge case where we couldn't find the closed event even though
+          // we're closed. This could happen if there was a lot of continuing
+          // discussion after an issue was closed. Skip these issues
+          // altogether.
+          console.log('Missing closed date for closed issue:', issue.url);
```

Comment:
Let's remove this console.log(). I was imagining it would be useful during testing to ensure we had the right thresholds, but after that its fine to just skip them.

Reply:
Done.